salt/metalk8s/kubernetes/cni/calico/configured.sls

```diff
@@ -3,6 +3,13 @@
 
 {%- set kubernetes_service_ip = salt.metalk8s_network.get_kubernetes_service_ip() %}
 
+{#- Check that service MTU configured is inferior to the local workload interface one #}
+{%- set workload_local_mtu = salt.metalk8s_network.get_mtu_from_ip(grains.metalk8s.workload_plane_ip) %}
+{%- if pillar['networks']['mtu'] > workload_local_mtu %}
+  {{ raise('Trying to configure CNI with ' ~ pillar['metalk8s']['networks']['mtu']
+           ~ 'MTU but local workload interface MTU is inferior ' ~ workload_local_mtu) }}
+{%- endif %}
```

Comment:
Neat! Do you think we can add a similar check earlier in the bootstrap/`deploy_node` orchestrate?

Maybe consider adding these checks in `metalk8s.internal.preflight`?

Reply:
Yes I could add check earlier (maybe not in the preflight because to me preflight should be removed at some point, since it does not bring any value)

I will try to find a better place earlier I agree it fail a bit late here (at least for bootstrap because for new node it will fail when rendering the "bring new node to highstate" so not really late I think)

Reply:
Could be done later, and as you said, since we may want to do some refactoring around the pre-checks, would make sense to add as a task in #1999